website/docs/language/state/remote-state-data.mdx

```diff
@@ -9,17 +9,12 @@ description: >-
 
 [backends]: /language/settings/backends
 
-The `terraform_remote_state` data source retrieves the root module output values
-from some other Terraform configuration, using the latest state snapshot from
-the remote backend.
+The `terraform_remote_state` data source uses the latest state snapshot from the remote backend to retrieve the root module output values
+from some other Terraform configuration.
 
-This data source is built into Terraform, and is always available; you do not
-need to require or configure a provider in order to use it.
+You can use the `terraform_remote_state` data source without requiring or configuring a provider. It is always available through a built-in provider with the [source address](/language/providers/requirements#source-addresses) `terraform.io/builtin/terraform`. That provider does not include any other resources or data sources.
 
--> **Note:** This data source is implemented by a built-in provider, whose
-[source address](/language/providers/requirements#source-addresses)
-is `terraform.io/builtin/terraform`. That provider does not include any other
-resources or data sources.
+!> **Warning:** We recommend using the [`tfe_outputs` data source](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/data-sources/outputs) in the [Terraform Cloud/Enterprise Provider](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs) to access remote state outputs in Terraform Cloud or Terraform Enterprise. The `tfe_outputs` data source is more secure because it does not require full access to workspace state to fetch outputs. If you are using the [Terraform Cloud CLI integration](/docs/cli/cloud), you **must** use `tfe_outputs` because the integration is incompatible with `terraform_remote_state`.
```

Comment:
Hmm, I'm not sure if this merits a "!> Warning"; I'd suggest downgrading it to an "~> Important". 

```suggestion
~> **Important:** We recommend using the [`tfe_outputs` data source](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/data-sources/outputs) in the [Terraform Cloud/Enterprise Provider](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs) to access remote state outputs in Terraform Cloud or Terraform Enterprise. The `tfe_outputs` data source is more secure because it does not require full access to workspace state to fetch outputs.
```

Also, I suggest removing the final sentence because it's not quiiiiite accurate -- you can't configure a `terraform_remote_state` data source to use the cloud integration, but you CAN configure it with the `remote` backend to access the same workspace you might have wanted to access via the cloud integration. That's kind of complicated to express without giving the wrong idea, so, better to just omit that and leave the message at "use tfe_outputs if you're on TFC, it's nicer."

Reply:
gotcha!! thank you - I really appreciate both the correction and the explanation!